Code/C++/ModularArithmetic.cpp

```diff
@@ -0,0 +1,192 @@
+/* Program that performs ADDDTION,SUBTRACTION, MULTIPLICATION, DIVISION, POWER.
+so, we are taking inputs 'a' and 'b' for above operaitions and also given MOD 'm'.
+All of them have constraints 1 to 10^50000
+
+Explaining the main points- 
+
+1 .  For performing DIVISION (a/b) we have to calculate the multiplicative inverse of 'b' first. The value of x should be in { 1, 2, … m-1}, i.e., in the range of integer 
+modulo m. ( Note that x cannot be 0 as a*0 mod m will never be 1 )
+The multiplicative inverse of “a modulo m” exists if and only if a and m are relatively prime (i.e., if gcd(a, m) = 1).
+
+2.  As our inputs are larger (10^50000) we cannot store them using long long int or any other datatype, so we are going to store them in string and convert it into 
+long long int datatybe, after gradually and simulteneously calulating modulo operation.
+
+3.  For above,  According to Feramts Little Theorem  
+         a^(p-1) mod p   =  1, When p is prime.
+   From this, as of the problem, M is prime, express A^B mod M as follows: 
+          A^B mod M = ( A^(M-1) * A^(M-1) *.......* A^(M-1) * A^(x) ) mod M          
+   Where x is B mod M-1 and A ^ (M-1) continues B/(M-1) times
+   Now, from Fermat’s Little Theorem,  
+           A ^ (M-1) mod M = 1.
+           A^B mod M = ( 1 * 1 * ....... * 1 * A^(x) ) mod M
+   Hence mod B with M-1 to reduce the number to a smaller one and then use power() method to compute (a^b)%m. 
+            
+   
+TIME COMPLEXITY - 
```

Comment:
Add time complexity below the code.

Reply:
@OjusWiZard  Completed the requested changes, please review this again.